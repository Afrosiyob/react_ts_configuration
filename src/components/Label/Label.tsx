/* eslint-disable @typescript-eslint/strict-boolean-expressions */
import React, { ReactElement } from "react";
import cx from "classnames";

import classes from "./scss/style.module.scss";

interface IProps {
  title?: string | boolean;
  required: boolean;
  link?: string;
  icon?: JSX.Element;
  error: boolean;
  position?: "horizontal" | "vertical" | undefined;
  children: JSX.Element;
}

const Label = ({
  title,
  required,
  link,
  icon,
  error,
  position = "vertical",
  children,
}: IProps): ReactElement => (
  <div
    className={cx(
      classes.wrapper,
      classes[`wrapper--position-${position}`],
      error && classes["wrapper--error"]
    )}
    data-testid="ANT_LABEL_ID"
  >
    <div className={classes.header}>
      <div className={classes.headerLeft}>
        <div className={classes.title}>
          {required && <span>*</span>}
          {title}
        </div>
      </div>
      {(!!link || !!icon) && (
        <div className={classes.headerRight}>
          {!(link?.length === 0) && <div className={classes.link}>{link}</div>}
          {!!icon && <div className={classes.icon}>{icon}</div>}
        </div>
      )}
    </div>
    {!!children && <div className={classes.content}>{children}</div>}
  </div>
);

export default Label;
