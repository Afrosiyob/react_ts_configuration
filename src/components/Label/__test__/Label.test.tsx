import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import Label from "../Label";

describe("Test for Component 'AntLabel'", () => {
  it("should render", () => {
    render(
      <Label required={false} error={false} title="Label">
        <div> children </div>
      </Label>
    );

    const component = screen.getByTestId("ANT_LABEL_ID");

    expect(component).toBeInTheDocument();
  });

  it("should work props", () => {
    const testRenderer = renderer.create(
      <Label required={false} error={false} title="Label">
        <div> children </div>
      </Label>
    );

    const testInstance = testRenderer.root;

    expect(testInstance.props.required).toBe(false);
    expect(testInstance.props.error).toBe(false);
    expect(testInstance.props.title).toBe("Label");
  });
});

describe("Snapshot for component 'AntLabel'", () => {
  it("should match DOM", () => {
    const tree = renderer.create(
      <Label required={false} error={false} title="Label">
        <div> children </div>
      </Label>
    );

    expect(tree).toMatchSnapshot();
  });
});
