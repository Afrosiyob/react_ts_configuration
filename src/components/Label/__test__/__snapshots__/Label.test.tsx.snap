// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for component 'AntLabel' should match DOM 1`] = `
<div
  className="wrapper wrapper--position-vertical"
  data-testid="ANT_LABEL_ID"
>
  <div
    className="header"
  >
    <div
      className="headerLeft"
    >
      <div
        className="title"
      >
        Label
      </div>
    </div>
  </div>
  <div
    className="content"
  >
    <div>
       children 
    </div>
  </div>
</div>
`;
